### PR TITLE
refactor: move wal structs and traits to wal crate

### PR DESCRIPTION
**Pull request by tisonkun**
## Rationale

Ref - https://github.com/CeresDB/ceresdb/pull/1231#discussion_r1337916665.

## Detailed Changes

See the commit messages.

## Test Plan

Code refactor. All existing tests should pass.

### CODE REVIEW

File: wal/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+use crate::{
+    message_queue_impl::config::KafkaWalStorageConfig, rocks_impl::config::RocksDBWalStorageConfig,
+    table_kv_impl::config::ObkvWalStorageConfig,
+};
+
+/// Options for wal storage backend
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(tag = "type")]
+pub enum WalStorageConfig {
```
Comment (jiacai2050) [inline wal/src/config.rs:25]: This is enum is already in wal module, I suggest we name it `StorageConfig` directly.

File: wal/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    RocksDB(Box<RocksDBWalStorageConfig>),
```
Comment (jiacai2050) [inline wal/src/config.rs:26]: Following same rules, I suggest remove `Wal` from those configs too.

**Comment by tisonkun** [comment]
@jiacai2050 Thanks for your review! Updated.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
@jiacai2050 shall we merge this one and proceed #1231, or we're still waiting for someone's review?

**Comment by jiacai2050** [comment]
Merged, thanks.

**Pull request merged**